### PR TITLE
Optimize DashboardHeader rendering by using selector

### LINKED ISSUE #42885: Optimize rendering of DashboardHeader
Screenshot 2024-05-20 at 15 56 37

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/42885

### Description

Dashboard header requires metadata just to understand if model actions are enabled, so we optimize its usage here with selector

### CODE REVIEW

File: frontend/src/metabase/dashboard/selectors.ts [hunk 1]
```diff
@@ -530,3 +538,19 @@ export const getIsNightMode = createSelector(
...
 );
+
+export const getHasModelActionsEnabled = createSelector(
+  [getMetadata],
+  metadata => {
+    if (!metadata) {
+      return false;
+    }
+
+    const databases = metadata.databases as Record<DatabaseId, Database>;
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/selectors.ts:549]: metadata.databaseList()

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/dashboard/selectors.ts (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/selectors.ts:549]: @ranquild is databaseList somehow better? types are not complete and overall it's marked as a deprecated method

Comment (ranquild) [reply frontend/src/metabase/dashboard/selectors.ts:549]: It has the correct type. It's not `Database`

Comment (ranquild) [reply frontend/src/metabase/dashboard/selectors.ts:549]: but I'm fine leaving this as it

Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/selectors.ts]: changed, but types do not match, left ts-expect-error

**Pull request merged**